crypto/addresses/src/lib.rs

```diff
@@ -295,4 +600,53 @@ mod tests {
         assert_eq!(Err(AddressError::BadChecksum), address);
         // cspell:enable
     }
+
+    use js_sys::Object;
+    use wasm_bindgen::{JsValue, __rt::IntoJsResult};
+    use wasm_bindgen_test::wasm_bindgen_test;
+    use workflow_wasm::{extensions::ObjectExtension, serde::from_value, serde::to_value};
+
+    #[wasm_bindgen_test]
+    pub fn test_wasm_serde_constructor() {
+        let str = "kaspa:qpauqsvk7yf9unexwmxsnmg547mhyga37csh0kj53q6xxgl24ydxjsgzthw5j";
+        let a = Address::constructor(str);
+        let value = to_value(&a).unwrap();
+
+        assert_eq!(JsValue::from_str("string"), value.js_typeof());
+        assert_eq!(value, JsValue::from_str(str));
+        assert_eq!(a, from_value(value).unwrap());
+    }
+
+    #[wasm_bindgen_test]
+    pub fn test_wasm_js_serde_object() {
+        let expected = Address::constructor("kaspa:qpauqsvk7yf9unexwmxsnmg547mhyga37csh0kj53q6xxgl24ydxjsgzthw5j");
+
+        use web_sys::console;
+        console::log_4(&"address: ".into(), &expected.version().into(), &expected.prefix().into(), &expected.payload().into());
+
+        let obj = Object::new();
+        obj.set("version", &JsValue::from_str("PubKey")).unwrap();
+        obj.set("prefix", &JsValue::from_str("kaspa")).unwrap();
+        obj.set("payload", &JsValue::from_str("qpauqsvk7yf9unexwmxsnmg547mhyga37csh0kj53q6xxgl24ydxjsgzthw5j")).unwrap();
+
+        assert_eq!(JsValue::from_str("object"), obj.js_typeof());
+
+        let obj_js = obj.into_js_result().unwrap();
+        let actual = from_value(obj_js).unwrap();
+        assert_eq!(expected, actual);
+    }
+
+    #[wasm_bindgen_test]
+    pub fn test_wasm_serde_object() {
+        use wasm_bindgen::convert::IntoWasmAbi;
+
+        let expected = Address::constructor("kaspa:qpauqsvk7yf9unexwmxsnmg547mhyga37csh0kj53q6xxgl24ydxjsgzthw5j");
+        let wasm_js_value: JsValue = expected.clone().into_abi().into();
+
+        // use web_sys::console;
```

Comment:
This can be useful in debug purpose, so it's the only one way how to print something from wasm_bindgen_test, by default it's unnecessary, that's why I kept it like that